### ISSUE TITLE
Ambiguous import not detected

### ISSUE THREAD

**Issue by sander2** [rust-lang/rust#97584]
I tried this code ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=0c43489bbcbd12ba1144780a45f784be)):

```rust
use a::*;
mod a {
    pub use b::*;
    mod b {
        pub const SOME_VALUE: u32 = 1;
    }

    pub use c::*;
    mod c {
        pub const SOME_VALUE: u32 = 2;
    }
}
fn main() {
    assert_eq!(SOME_VALUE, 1);
}

```

I expected to see this happen: an error like `error[E0659]: SOME_VALUE is ambiguous`

Instead, this happened: the code compiles and runs to completion.

If you put the `use a::*` _after_ the `mod a`, the expected error is thrown.

### Meta
This happens both on stable and nightly.

**Comment by SNCPlay42** [comment]
This [fails to compile](https://rust.godbolt.org/z/1WY7jMjEa) (after adapting to 2015 edition) on 1.14 and compiles on 1.15 and later.

1.15 [made changes](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1150-2017-02-02) to the name resolution rules as described in [RFC 1560](https://github.com/rust-lang/rfcs/blob/master/text/1560-name-resolution.md). The relevant part seems to be:

>### [Multiple unused imports](https://github.com/rust-lang/rfcs/blob/master/text/1560-name-resolution.md#multiple-unused-imports)
>
>A name may be imported multiple times, it is only a name resolution error if that name is used. E.g.,
>
>```rust
>mod foo {
>    pub struct Qux;
>}
>
>mod bar {
 >   pub struct Qux;
>}
>
>mod baz {
>    use foo::*;
>    use bar::*; // Ok, no name conflict.
>}
>```
>
>In this example, adding a use of Qux in baz would cause a name resolution error.

Attempting to lookup `a::SOME_VALUE` [does indeed error](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=c88c5714900631b21917f223d81237da). However the compiler appears to fail to consider what happens if a name that is ambiguous due to multiple imports is glob-imported into *another* module. As far as I can tell, it should still be ambiguous, but instead it appears the first import of the name "wins".

@rustbot label A-resolve regression-from-stable-to-stable

(Ancient regression, I know, but this issue should probably have a priority assigned anyway)

**Comment by apiraino** [comment]
Assigning priority as discussed in the [Zulip thread](https://rust-lang.zulipchat.com/#narrow/stream/245100-t-compiler.2Fwg-prioritization.2Falerts/topic/.2397584.20Ambiguous.20import.20not.20detected) of the Prioritization Working Group.

@rustbot label -I-prioritize +P-medium

**Comment by SNCPlay42** [comment]
I think this is a duplicate of #47525.

**Comment by apiraino** [comment]
@SNCPlay42 would you prefer to close this in favor of #47525 (and working on that one instead)?

**Comment by sander2** [comment]
> I think this is a duplicate of #47525.

I think you're right. I'm ok with closing this issue then. Just note that the linked issue has different tags (`enhancement` vs `bug`, no priority, and no regression tag) which may have contributed in it being an open issue for 4 years

**Comment by apiraino** [comment]
agreed, will update #47525 accordingly. Thanks!

**Comment by apiraino** [comment]
Closing this in favor of #47525
